langchain/schema.py

```diff
@@ -378,63 +518,104 @@ async def aget_relevant_documents(self, query: str) -> List[Document]:
         """Get documents relevant for a query.
 
         Args:
-            query: string to find relevant documents for
+            query: String to find relevant documents for
 
         Returns:
             List of relevant documents
         """
 
 
 # For backwards compatibility
-
-
 Memory = BaseMemory
 
 T = TypeVar("T")
 
 
 class BaseLLMOutputParser(Serializable, ABC, Generic[T]):
+    """Abstract base class for parsing the outputs of a language model to match a given
+    structure/format.
+    """
+
     @abstractmethod
     def parse_result(self, result: List[Generation]) -> T:
-        """Parse LLM Result."""
+        """Parse a list of candidate model Generations into a specific format.
+
+        Args:
+            result: A list of Generations to be parsed. The Generations are assumed
+                to be different candidate outputs for a single model input.
+
+        Returns:
+            Structured output.
+        """
 
 
 class BaseOutputParser(BaseLLMOutputParser, ABC, Generic[T]):
     """Class to parse the output of an LLM call.
 
     Output parsers help structure language model responses.
-    """
+
```

Comment:
something about how this is differnt than BaseLLMOutputParser

Reply:
tbh do we want most folks to look at BaseLLMOutputParser? kinda feels like this is the much more generic thing and maybe we somehow hide the BaseLLMOutputParser more